packages/react-components/react-datepicker-compat/src/components/DatePicker/useDatePicker.tsx

```diff
@@ -101,7 +100,7 @@ const defaultParseDateFromString = (dateStr: string) => {
  * @param props - props from this instance of DatePicker
  * @param ref - reference to root Input slot
  */
-export const useDatePicker_unstable = (props: DatePickerProps, ref: React.Ref<HTMLElement>): DatePickerState => {
+export const useDatePicker_unstable = (props: DatePickerProps, ref: React.Ref<HTMLInputElement>): DatePickerState => {
```

Comment:
This is also a breaking change.

Reply:
I'm not sure if this counts as breaking, HTMLInputElement extends HTMLElement. This means that we are just adding functionality correct? Though in usability, it will complain for users unless it's casted. So, it's kind of breaking? 

...let's say it's breaking since it breaks functionality?